### PR TITLE
chore: Add update script and explanation

### PR DESCRIPTION
**Pull request by dsimansk**
# Changes

- chore: Add update script and explanation

Note: Knobot will be enabled afterwards this PR is merged and script present.

/cc @rhuss

### CODE REVIEW

**Comment by dsimansk** [comment]
Also depends on to work as expected https://github.com/knative-sandbox/kn-plugin-source-kamelet/pull/201.

File: README.md [hunk 1]
```diff
@@ -2,4 +2,23 @@
 
 Repository for sharing Knative resources to ArtifactHub.
 
-https://artifacthub.io/docs/topics/repositories/knative-client-plugins/
\ No newline at end of file
+https://artifacthub.io/docs/topics/repositories/knative-client-plugins/
+
+### Adding new artifact metadata
+
+The metadata stored in `plugins` directory are periodically indexed by ArtifactHub to list our plugins per `kn-client-plugin` type of AH.
```
Comment (rhuss) [inline README.md:9]: Would expand AH, though relatively clear that this means "Artifact Hub" (or even skip it)
```suggestion
The metadata stored in `plugins` directory is periodically indexed by Artifact Hub to list our plugins per `kn-client-plugin` type.
```

File: README.md [hunk 2]
```diff
@@ -2,4 +2,23 @@
...
+
+Step to add a new one:
+ - Create new directory named per GitHub repository name
```
Comment (rhuss) [inline README.md:12]:
```suggestion
 - Create a new directory named per GitHub repository name
```

File: README.md [hunk 3]
```diff
@@ -2,4 +2,23 @@
...
+ - Create `artifact-pkg.yml` per docs link above
```
Comment (rhuss) [inline README.md:13]: What is "docs link above" ? Maybe expand it here again.

File: README.md [hunk 4]
```diff
@@ -2,4 +2,23 @@
...
+ - Execute `./hack/update-codegen.sh`
+
+To add a new type of artifact a top-level directory should be created with appropriate name.
```
Comment (rhuss) [inline README.md:16]:
```suggestion
To add a new type of artifact, a top-level directory should be created with an appropriate name.
```

File: README.md [hunk 5]
```diff
@@ -2,4 +2,23 @@
...
+
+### Updating `README.md` files
+
+There's a script to automatically download current `main` README files per plugin directory. It's periodically executed by Knobot action.
```
Comment (rhuss) [inline README.md:20]:
```suggestion
There's a script to automatically download the current `main` README files per plugin directory. It's periodically executed by a Knobot action. 
```

Comment (rhuss) [reply README.md:20]: Maybe add a link directly to the Knobot action. Not everybody knows, what "Knobot" is.

**Review by rhuss — COMMENTED** [review]
Some proof read corrections, otherwise fine. Thanks!

**Comment by dsimansk** [comment]
> Some proof read corrections, otherwise fine. Thanks!

@rhuss ready for the next round. 🙏

**Review by rhuss — COMMENTED** [review]
Thanks!

/approve
/lgtm

**Pull request merged**